app/views/comments/_form.html.erb

```diff
@@ -0,0 +1,14 @@
+<h3>Post a Comment</h3>
+<%= form_for :comment do |f| %>
```

Comment:
This should be @comment, but you will probably need `[@article, @comment]` as comment is a relation of article

Reply:
You'll also need to instantiate @comment in your articles_controller

Reply:
Hi Kris, I have @comment defined in a CommentsController so does it need to be in the articles_controller aswell?

Reply:
The comment controller will only be called when a comment fails validation.  I assume the comment form is inline on the article show view?

Reply:
I'm still struggling to get my head round this one.  Firstly I've inserted the form_for [@article, @comment], but it come up with the error -"First argument in form cannot contain nil or be empty".  Also I'm not sure how to instantiate @comment in the articles_controller.  I already have def create.. & def new.. in the comments controller so where do you intsantiate it? under  def show... ?

Reply:
I think I've worked it out Kris. I have done @comment = Comment.new within the def show... method and the error has gone and articles & comments now both showing.  I'll start working on the tests and send a PR when ready. Cheers.